app/ide-desktop/lib/content/src/index.ts

```diff
@@ -119,104 +117,134 @@ function displayDeprecatedVersionDialog() {
 }
 
 // ========================
-// === Main Entry Point ===
+// === Main entry point ===
 // ========================
 
 interface StringConfig {
     [key: string]: StringConfig | string
 }
 
-class Main {
-    async main(inputConfig: StringConfig) {
-        const config = Object.assign(
-            {
-                loader: {
-                    wasmUrl: 'pkg-opt.wasm',
-                    jsUrl: 'pkg.js',
-                    assetsUrl: 'dynamic-assets',
-                },
-            },
-            inputConfig
-        )
-
-        const appInstance = new app.App({
-            config,
-            configOptions: contentConfig.OPTIONS,
-            packageInfo: {
-                version: BUILD_INFO.version,
-                engineVersion: BUILD_INFO.engineVersion,
+// This is intended to be reassigned.
+// eslint-disable-next-line no-restricted-syntax
+let dropPreviousInstance = () => {
+    // Intentionally a no-op when there is no instance to drop.
+}
```

Comment:
I don't really understand why we are using such hacks. Instead of that a class can be created with fields and methods. One of the fields can be `appInstance: App | null` and this can be just a method of the class. Creating global functions that we will override from other functions is super hard to follow and I don't think we should ever do anything like that. TBH I hate JS that it allows such things.

(the same applies to the code below)

Reply:
hm, maybe this should have been `null` if there's no current app instance
i figured this would be conceptually better, since the old `appInstance` shouldn't be needed for anything.
but of course i have no problems with turning it into a method instead.

but i think for now i'll just pass a regular object first - it should be simple enough to change into a class if that's preferred anyway

Reply:
actually never mind, yeah it's probably a better idea to use a class. not sure if there's a better solution to avoiding polluting window than passing the `Main` instance to the authorization app though